### ISSUE TITLE
Java client lib: use stronger types for get/mark/failed experiment methods

### ISSUE THREAD

**Issue by jerith666** [CarnegieLearningWeb/educational-experiment-service-client-libs#18]
In `ExperimentServiceAPI`, instead of an arbitrary `Map<String,String>`, these methods should take a more structured / strongly-typed argument:

```
Call<List<ExperimentsResponse>> getAllExperiments(@Body Map<String, String> body);
Call<MarkExperimentPoint> markExperimentPoint(@Body Map<String, String> body);
Call<FailedExperiment> failedExperimentPoint(@Body Map<String, String> body);
```

**Comment by dhavalocked** [comment]
Fixed in #21

**Comment by jerith666** [comment]
#21 addressed this.
